### DISCUSSION TITLE
“Search for previous notes ” sometime failed

### DISCUSSION THREAD

**Discussion by Tback1** [coddingtonbear/obsidian-web#167] [Bug Report]
### Describe the bug

The web page has been cut to the warehouse, but the duplicate detection fails from time to time, some web pages work normally, and some web pages are refreshed, and the duplicate detection is not normal

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/559d516d-6d8f-42eb-bbc4-8916ede39f3d)

[test by the web](https://coddingtonbear.github.io/obsidian-local-rest-api)
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/c87149ca-32cc-408b-9e6a-a534f1ec0e78)

### To Reproduce

1. this is happened after updated `3.2.1` but the `Local Rest API` still `2.0.4` so the error budge show again,then I update OWeb to `2.1.0`, it come this error
2. the url could test
- `https://sspai.com/post/86447`
- `https://utgd.net/article/20582`

by the way the 2.1.0 Clipping some times failed too, need do it again

### Operating System

win10 LTSC

### Browser

Chrome

### Browser Version

 120.0.6099.225

local rest api 2.1.0
Obsidian Web 3.2.4

**Comment by coddingtonbear** [comment]
I'm afraid that it's basically certain that that "empty" response you've highlighted is correct -- whatever it's searching for is not in your vault.

The more likely situation is that the URL you're searching for isn't exactly the same as the URL you're landing on, and I can't see enough from the detail above to point you at the actual cause.  I've seen this happen in situations as simple as a missing (or added) extra slash at the end of a URL, to as complex as when a site uses "SEO" techniques like showing human-readable (but unpredictable) text on outside links, but lands you on a different page when you actually navigate to it.

In all cases, though, there's not remotely enough information from the above to even hazard a guess about what's going on in your particular case, though.  My guess, from just glancing at your above screenshots, is that the URL you're searching for isn't actually in your vault, and instead there's a very similar URL there that you're interpreting as being the same.

As an aside: I'm not sure what I'm supposed to be gathering from that second screenshot.  That particular `GET` route isn't used as part of duplicate detection.

**Reply by Tback1** [reply]
I write down the URL in the discuss to test. the URL is just clipping into the Vault
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/020c7bd1-0579-4305-b151-fc30aa6303d0)

some time it could 
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/86a0bfca-d9c1-40ba-a14e-3a0d0a08dd27)

I'm doubt this question is by Local Rest API

**Reply by coddingtonbear** [reply]
Can you see what URL it's searching for in that search request?  You might be able to shed more light if you upload a bug report export.  You can find that from the red "bug"-shaped icon on the extension options page.  In this case, we'd need you to:

1. Go to that URL and observe that it doesn't find the match.
2. Immediately go to the bug report export (quickly is important!)
3. Check the box labeled "Include log entries from the previous fifteen minutes"
4. Export the bug report logs.
5. Upload them here.

**Reply by Tback1** [reply]
现象变为保存了某个网页后 必须刷新页面，否则插件不会提示该网页重复

The phenomenon becomes that after a web page is saved, the page must be refreshed, otherwise the plugin will not prompt the page to be duplicated

Local Rest API updated to 2.1.1

**Reply by Tback1** [reply]
@coddingtonbear 

我可以确信，这的确是个bug了，可以复现，之前我怀疑是 `omniserach` 有影响，但现在我删除了该拓展，问题依旧

I'm sure it's a bug and can be reproduced, I suspected 'omniserach' was having an effect, but now I've removed the extension and the problem is still there

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/9ceb5b50-2220-45b5-9db1-55a3646f953c)

## log
[obsidian-local-rest-api.bug_report.json](https://github.com/coddingtonbear/obsidian-web/files/14402033/obsidian-local-rest-api.bug_report.json)

**Reply by coddingtonbear** [reply]
It's very difficult for me to piece together what exactly might be going on without access to your vault, but I can see the following in the last few messages in your logs:

- We search for "Page Notes" for the URL `https://utgd.net/signup`, and are not finding matches.  **If you believe this is in error**, send me the content of the note that you're expecting it to match.
- We also search for "mentions" for the URL you're on using the regular expression `https://utgd\\.net/signup(?=\\s|\\)|$)` and still  aren't finding matches.  Again, **if you believe this is in error**, send me the content of the note you're expecting it to match.

I have to say that I'd be pretty astonished if this wasn't working in this case since I use this feature regularly; so I definitely suspect that your notes just don't quite match what we're searching for and are in fact just subtly different from what we're searching for, but I've been surprised before.

**Reply by Tback1** [reply]
in this case I need it to finding 'https://utgd.net/article/20484' and I checked the log ,it search for so many times 'https://utgd.net/article/20484' and responded OK,but I don't know why it does't tell me it's duped?

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/de48e27c-9235-4a3c-89b5-f68552b01280)

**Comment by coddingtonbear** [comment]
> 现象变为保存了某个网页后 必须刷新页面，否则插件不会提示该网页重复
> 
> The phenomenon becomes that after a web page is saved, the page must be refreshed, otherwise the plugin will not prompt the page to be duplicated
> 
> Local Rest API updated to 2.1.1

This is the intended behavior.  The intention of the feature is to remind you of what notes you have for a page *when you return there again*.

**Reply by Tback1** [reply]
but some page refreshed still can't get find it as the last post I send the Log，the page in the Vault almost 3 months

**Reply by Tback1** [reply]
I'd like it before save and not need to refresh, then could find the dupe again

**Comment by Tback1** [comment]
![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/40f4a4bb-d926-4df7-a342-0b81e36fae0b)

@coddingtonbear 
The interesting thing is that **if I open this URL from a link to another page**, then there is an issue where duplicates cannot be found. 
And I directly **type the URL in the address bar and open the page**, and it can find that the page is duplicated. It's really exquisite

事情有趣的地方在于，如果我从其他页面的链接打开这个网址，那么就会出现找不到重复项的问题。
而我直接在地址栏输入网址并打开该页面，它又能找到该页面重复。真是极其精分

## Log:
[obsidian-local-rest-api.bug_report (1).json](https://github.com/coddingtonbear/obsidian-web/files/14485900/obsidian-local-rest-api.bug_report.1.json)

**Comment by coddingtonbear** [comment]
I'm surprised to see that you've closed this!  The only way I could imagine the recent release fixing this is if the problem _wasn't_ that the search was failing, but instead just that the link wasn't present in your notes at all (because of the bug mentioned earlier).

**Reply by Tback1** [reply]
I'm closing this issue because the issue has been resolved. But the reason is not what you guess.

In that reply, I mentioned [several phenomena that were related to another issue](https://github.com/coddingtonbear/obsidian-web/discussions/172#discussioncomment-8698209). But your reply was interrupted halfway through, and I didn't have much time to think about it last night. **The whole problem is likely that the title of the newly opened page is not obtained in the first time in the chrome**, resulting in a series of strange phenomena. However, it seems that with the cooperation of the latest ob-web and Local Rest API updates, it has been resolved. So I closed this issue, at least I don't have the above issue with the latest version here for the time being

It's nice that this torturous set of problems has been solved

---

> but instead just that the link wasn't present in your notes at all (because of the bug mentioned 
earlier).

**I don't know why you think this,if you mentioned those two replies**

https://github.com/coddingtonbear/obsidian-web/discussions/167#discussioncomment-8670743

https://github.com/coddingtonbear/obsidian-web/discussions/167#discussioncomment-8589706
